ui/perfherder/graphs/GraphsContainer.jsx

```diff
@@ -87,6 +90,39 @@ class GraphsContainer extends React.Component {
     }
   }
 
+  /**
+   * By default, Victory chart will place dots at the very ends of the graphs, which
+   * cuts them off. This function takes into account the data and dot size to compute
+   * a domain that will ensure the dots are all charted within view.
+   *
+   * Note that the domainPadding provided by Victory pads the data incorrectly, and
+   * skews the positioning of the graph.
+   */
+  getDomainX() {
+    const { zoomDomain, scatterPlotData } = this.state;
+
+    // Due to Bug 1676498 some data points can appear in the future. Guard against
+    // this by accepting dates in the future.
+    const latestDate = scatterPlotData.reduce(
+      (max, p) => (p.x > max ? p.x : max),
+      new Date(),
+    );
+
+    const paddingInMilliseconds =
+      // Figure out the length of the graph in Milliseconds.
+      (Number(latestDate) - Number(zoomDomain.minX)) *
+      // Multiply by the ratio of 1 dot in terms of the width of the chart. The 1.4
+      // factor is used here since this is done once for each dot, and the factor
+      // needs to be increased to fully show the dot on the screen. This number
+      // was determined visually.
+      ((DOT_SIZE * 1.4) / CHART_WIDTH);
+
+    return {
+      min: new Date(Number(zoomDomain.minX) - paddingInMilliseconds),
```

Comment:
It's been a while since I've worked on this code, but I'm wondering if it will matter that this is a localized date instead of UTC, like we're using elsewhere for this graph.

Reply:
Thanks for the thorough explanation!